app/controllers/idv/capture_doc_status_controller.rb

```diff
@@ -11,26 +11,33 @@ def show
     private
 
     def document_capture_session_poll_render_result
-      return { plain: 'Unauthorized', status: :unauthorized } unless flow_session
```

Comment:
Toyed with a few refactoring ideas. Not sure if it's any better, and a bit extensive for what we're trying to change here.

```rb
module Idv
  class CaptureDocStatusController < ApplicationController
    before_action :confirm_two_factor_authenticated

    respond_to :json

    def show
      render(
        json: ({ redirect: idv_session_errors_throttled_url } if status == :too_many_requests),
        status: status,
      )
    end

    private

    def status
      @status ||= begin
        if !flow_session || !document_capture_session
          :unauthorized
        elsif document_capture_session.cancelled_at
          :gone
        elsif throttled?
          :too_many_requests
        elsif session_result.blank?
          :accepted
        elsif !session_result.success?
          :unauthorized
        else
          :ok
        end
      end
    end

    def flow_session
      user_session['idv/doc_auth']
    end

    def session_result
      return @session_result if defined?(@session_result)
      @session_result = document_capture_session.load_result ||
                        document_capture_session.load_doc_auth_async_result
    end

    def document_capture_session
      return @document_capture_session if defined?(@document_capture_session)
      session_uuid = flow_session[:document_capture_session_uuid]
      @document_capture_session = DocumentCaptureSession.find_by(uuid: session_uuid)
    end

    def throttled?
      Throttler::IsThrottled.call(document_capture_session.user_id, :idv_acuant)
    end
  end
end
```

Reply:
I like the simplicity of new `show` method. Maybe another way to simplify would be to put the status in the body so so there's less branching? (even if the client's don't use it)

```ruby
   def show
      render(
        json: {
          status: status,
          redirect: status == :too_many_requests ? idv_session_errors_throttled_url : nil,
        },
        status: status,
      )
    end
```

Reply:
I like it 👍  bcc5d85